packages/typespec-ts/src/modular/helpers/clientHelpers.ts

```diff
@@ -57,10 +60,7 @@ export function getClientParameters(
   const skipCredentials = (p: SdkParameter | SdkHttpParameter) =>
     p.kind !== "credential";
   const skipMethodParam = (p: SdkParameter | SdkHttpParameter) =>
-    p.kind !== "method" ||
-    (p.kind === "method" &&
-      p.isApiVersionParam &&
-      _client.parameters.find((p) => p.isApiVersion));
+    p.kind !== "method" || (p.kind === "method" && p.isApiVersionParam);
```

Comment:
need to verify 2 cases:
1. every operation has apiVersion, but not using versioned decorators.
2. not every operation has apiVersion, but are using versioned decorators.